include/xnvme_be_vfio.h

```diff
@@ -12,14 +12,17 @@
 #include <xnvme_be.h>
 #include <xnvme_queue.h>
 
+#define XNVME_BE_VFIO_NQUEUES_MAX 16
+
 struct xnvme_be_vfio_state {
 	struct nvme_ctrl *ctrl;
 	unsigned long long qidmap; // Queue identifier bit map
 
 	struct nvme_sq *sq_sync; // Submission queue for synchronous IOs
 	struct nvme_cq *cq_sync; // Completion queue for synchronous IOs
 
-	uint8_t _rsvd[94];
+	int efds[XNVME_BE_VFIO_NQUEUES_MAX]; // completion event FDs
```

Comment:
`vfio_set_irq()` is used to associate eventfds with a range of *vectors* - not queues. Many devices have fewer vectors available than the number of queues they support (i.e., the Optane I have at hand only has 9 MSI-X vectors).

You can get this number from the ``ctrl.pci.dev.irq_info``.